server/src/main/java/org/opensearch/env/EnvironmentSettingsResponse.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.env;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.transport.TransportResponse;
+import org.opensearch.common.settings.WriteableSetting;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Environment Settings Response for Extensibility
+ *
+ * @opensearch.internal
+ */
+public class EnvironmentSettingsResponse extends TransportResponse {
+    private Map<Setting<?>, Object> componentSettingValues;
+
+    public EnvironmentSettingsResponse(Settings environmentSettings, List<Setting<?>> componentSettings) {
+        Map<Setting<?>, Object> componentSettingValues = new HashMap<>();
+        for (Setting<?> componentSetting : componentSettings) {
+
+            // Retrieve component setting value from enviornment settings, or default value if it does not exist
+            Object componentSettingValue = componentSetting.get(environmentSettings);
```

Comment:
`componentSettingValue` could be `null`?

Reply:
@reta This file looks very different from a later PR.  Will update this PR with those changes